### PR TITLE
[Network Proxy] Allow both grpc and http-connect mode to be toggled in kube-up

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>

/kind cleanup

**What this PR does / why we need it**:

Allows us to selective trigger HTTPConnect or GRPC mode for network proxy from the CLI flag. This enables the CI job for network proxy to run on both modes.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/cc @caesarxuchao 
/assign @cheftako

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -805,7 +805,7 @@ contexts:
   name: webhook
 EOF
   fi
-  if [[ "${ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}" == "true" ]]; then
+  if [[ "${ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}" == "grpc" ]]; then
```
Comment (BenTheElder) [inline cluster/gce/gci/configure-helper.sh]: this seems like a hack but ... sure :-)

Comment (BenTheElder) [reply cluster/gce/gci/configure-helper.sh]: not going to block this PR on this, but for literal strings prefer `'grpc'`, single quoted contents stay literal.

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -823,6 +823,27 @@ egressSelections:
...
+        udsName: /etc/srv/kubernetes/konnectivity-server/konnectivity-server.socket
+- name: master
+  connection:
+    proxyProtocol: Direct
+- name: etcd
+  connection:
+    proxyProtocol: Direct
+EOF
+  elif [[ "${ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}" != "false" ]]; then
+    echo "ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE must be set to either http-connect or grpc"
```
Comment (BenTheElder) [inline cluster/gce/gci/configure-helper.sh]: or false?

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (Jefftree) [reply cluster/gce/gci/configure-helper.sh]: got rid of the hack :-)

File: cluster/gce/gci/configure-helper.sh [hunk 3]
```diff
@@ -1660,7 +1683,13 @@ function prepare-konnectivity-server-manifest {
   params+=("--uds-name=/etc/srv/kubernetes/konnectivity-server/konnectivity-server.socket")
   params+=("--cluster-cert=/etc/srv/kubernetes/pki/apiserver.crt")
   params+=("--cluster-key=/etc/srv/kubernetes/pki/apiserver.key")
-  params+=("--mode=grpc")
+  if [[ "${KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE:-grpc}" == 'http-connect' ]]; then
+    params+=("--mode=http-connect")
+  else
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh]: Please reverse the order (grpc first), add an else if for the http-connect and have a final else which gives an error and exits.
Eg. https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L830

**Review by cheftako — COMMENTED** [review]
/lgtm

**Pull request merged**